### ISSUE TITLE
BitsharesPriceFeed Unit Test

### ISSUE THREAD

**Issue by thehapax** [Codaone/DEXBot#592]

**Comment by bitProfessor** [comment]
https://github.com/bitProfessor/DEXBot/tree/devel/tests/pricefeeds
hi @thehapax ,#592   It looks like three test cases failed. I'm not sure why. Can you see?

**Comment by thehapax** [comment]
Which 3 test cases failed?

**Comment by bitProfessor** [comment]
http://htmlpreview.github.io/?https://github.com/bitProfessor/DEXBot/blob/devel/tests/pricefeeds/report.html
@thehapax

**Comment by thehapax** [comment]
The error " AssertionError: assert 'BTS' != 'BTS'" should not happen if the other tests pass. I still cannot see the updates on your master branch, so i don't have any code to reference except your unit test. It should, however, match the code from the most recent Codaone/devel which is here:

https://github.com/Codaone/DEXBot/blob/devel/dexbot/pricefeeds/bitshares_feed.py

**Comment by thehapax** [comment]
In your test code, for starters you have != instead of ==. In  buy orders you have ==, wherease in sell ordres you have !=
```
 def test_get_market_buy_orders(self):
        buy_orders = self.pf.get_market_buy_orders(depth=10)
        for o in buy_orders:
            assert o['base']['symbol'] == self.market['base']['symbol']

    def test_get_market_sell_orders(self):
        sell_orders = self.pf.get_market_sell_orders(depth=10)
        for o in sell_orders:
            assert o['base']['symbol'] != self.market['base']['symbol']
```

**Comment by bitProfessor** [comment]
1、fixed it.
2、I bought a new computer and I'm switching. git pull later

**Comment by bitProfessor** [comment]
hi @thehapax :
1、I tested it with 2 input data：USD:BTS,CNY:BTS
2、All passed the test （USD:BTS ）
http://htmlpreview.github.io/?https://github.com/bitProfessor/DEXBot/blob/dev/tests/pricefeeds/usd_bts_report.html
3、5 failed, 9 passed(CNY:BTS)
http://htmlpreview.github.io/?https://github.com/bitProfessor/DEXBot/blob/dev/tests/pricefeeds/cny_bts_report.html

**Comment by bitProfessor** [comment]
hi @thehapax :
1、in docker :
http://htmlpreview.github.io/?https://github.com/bitProfessor/DEXBot/blob/dev/tests/pricefeeds/doocker_report.html
2、Maybe it's py-bitshares's problem.

**Comment by bitfag** [comment]
@bitProfessor yes, this https://github.com/bitshares/python-bitshares/issues/234#issuecomment-502697543

Update your tests to devel branch where shared instance is used, this solves the problem.

**Comment by bitProfessor** [comment]
Thaks


| |
Professor
|
|
bitprofessor@163.com
|
签名由网易邮箱大师定制


On 06/17/2019 22:08，Vladimir Kamarzin<notifications@github.com> wrote：

@bitProfessor yes, this bitshares/python-bitshares#234 (comment)

Update your tests to devel branch where shared instance is used, this solves the problem.

**Comment by thehapax** [comment]
Hi bitProfessor, happy to see this complete. what branch is it on? (don't see a pull)
